### DISCUSSION TITLE
BypassPluginExecution not responding as expected

### DISCUSSION THREAD

**Discussion by TomWinton** [microsoft/PowerPlatform-DataverseServiceClient#228] [Q&A] [answered]
Hallo! 

2 things.  Using 0.5.10

ServiceClient.BypassPluginExecution is not present

//edited 
Asynchronous operations don't fire when I am including the bypass plugins in the request.

**Will complete operation but async workflows don't fire**

```
public static void createReq(ServiceClient Service, Entity entity)
        {
            CreateRequest request = new CreateRequest()
            {
                Target = entity            
            };      
           request.Parameters.Add("BypassCustomPluginExecution", true);
            try
            {          
                Service.ExecuteOrganizationRequest(request, "", true);
            }
            catch (Exception ex)
            {
            }
          }
```
        

But Service.Create(entity) with the same ServiceClient will execute asynchronous operations

**Comment by MattB-msft** [comment, marked as answer]
Hi @TomWinton , 
Thanks for your question... what you are seeing is by design,  use of the bypasspluginexecution is intended to be non-standard, and thus it's something that you do need to add code to use. 

If you form a Create Request directly and call any of the .Execute* methods, the system will process the bypass and bypass plugins where it is allowed.  You can also use the helpers provide in the system to support calling a create operation while passing bypass plugins ( .CreateRecord ) for example. 

The baseline IOrganizaitonService interface methods such as .Create do not support that form of modification and we currently do not intend to expose it. 

In the CrmServiceClient we added a class level switch as we were not able to modify the contracts of the helper methods w/out breaking compatibility.  A class level switch is not an ideal configuration as the switch setting is effective across threads\operations when it is intended to be an operation level control.   This was corrected in the dataverse serviceclient where were able to add operation level overrides to our current helpers and extensions. 

thanks 
MattB

**Reply by TomWinton** [reply]
Thankyou, i might still be misunderstanding something, I wanted to double check, from: https://docs.microsoft.com/en-us/powerapps/developer/data-platform/bypass-custom-business-logic#frequently-asked-questions-faq 

It states:
Asynchronous logic is not bypassed. Asynchronous logic doesn't significantly contribute to the cost of processing the records, therefore it is not by passed by this parameter.

Does this mean that the answer in this FAQ doesn't accurately reflect the behaviour or am I barking up the wrong tree?

**Reply by JimDaly** [reply]
@TomWinton 
I think you are barking up the wrong tree. :-)

@MattB-msft  is describing how to use the `BypassCustomPluginExecution` parameter with the DataverseServiceClient, which is the correct issue for this forum.

Your question appears to be about the behavior on the server. Documentation for the parameter in [Bypass Custom Business Logic](https://docs.microsoft.com/en-us/powerapps/developer/data-platform/bypass-custom-business-logic) states that asynchronous logic should not be bypassed when using this parameter.

If you are seeing that asynchronous logic (async workflows) is being by-passed, then either the documentation or the behavior of the parameter is not correct.  I wrote the documentation based on the design with the team, so I believe the behavior of the parameter on the server is not correct. 

I recommend that you contact support with a repro and we can understand what needs to be changed.

**Comment by TomWinton** [comment]
Roger!

**Comment by thenerdynerd** [comment]
Can you give an example of "You can also use the helpers provide in the system to support calling a create operation while passing bypass plugins ( .CreateRecord ) for example." we currently have a function app with DI that allows for bulk imports so to save time we want to bypass executions?

That would help my understanding. At the moment we are using CreateAsync operations and we have seen some flows that are firing that we would like to stop from happening for our import process.
